### ISSUE TITLE
[feat] Add prefix sharing

### ISSUE THREAD

**Issue by felipemello1** [joecummings/r1-zero#47]
Relevant for vllm and RefActor: creates special mask and avoids recomputing logits for repeated prompts: https://github.com/frankxwang/dpo-prefix-sharing
